samples/secure_services/src/main.c

```diff
@@ -35,8 +43,9 @@ static void print_random_number(uint8_t *num, size_t len)
 
 static int read_ficr_word(uint32_t *result, const volatile uint32_t *addr)
 {
+	void *cast_addr = (void*)addr;
 	printk("Read FICR (address 0x%08x):\n", (uint32_t)addr);
-	int ret = spm_request_read(result, (uint32_t)addr, sizeof(uint32_t));
+	int ret = soc_secure_mem_read(result, cast_addr, sizeof(uint32_t));
```

Comment:
Will this work with SPM? If not, shouldn't we remove this code if `BUILD_WITH_TFM` is not enabled?